File: qabelbox/src/main/java/de/qabel/qabelbox/fragments/CreateIdentityHeaderFragment.java
package de.qabel.qabelbox.fragments;

import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import de.qabel.qabelbox.R;

public class CreateIdentityHeaderFragment extends Fragment {

    private ViewGroup logoLayout;
    private ViewGroup initialLayout;
    private TextView tvInitial;
    private TextView tvName;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_create_identity_header, container, false);
        logoLayout = (ViewGroup) view.findViewById(R.id.logo_layout);
        initialLayout = (ViewGroup) view.findViewById(R.id.initial_layout);
        tvInitial = (TextView) view.findViewById(R.id.tv_initial);
        tvName = (TextView) view.findViewById(R.id.tv_name);
        updateUI(null);
        return view;
    }

    public void updateUI(String name) {

        if (name == null) {
            logoLayout.setVisibility(View.VISIBLE);
            initialLayout.setVisibility(View.INVISIBLE);
            return;
        }
        logoLayout.setVisibility(View.INVISIBLE);
        initialLayout.setVisibility(View.VISIBLE);
        tvName.setText(name);
        tvInitial.setText(getInitials(name));

        tvInitial.forceLayout();
        initialLayout.forceLayout();
    }

    private String getInitials(String name) {

        String[] names = name.split(" ");
        String result = "";
        for (String value : names) {
            if (result.length() < 2) {
                result += value.toUpperCase().charAt(0);
            }
        }
        return result;
    }
}

File: qabelbox/src/main/java/de/qabel/qabelbox/fragments/CreateIdentityHeaderFragment.kt
package de.qabel.qabelbox.fragments

import android.app.Fragment
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import de.qabel.qabelbox.R
import de.qabel.qabelbox.ui.extensions.setOrGone
import kotlinx.android.synthetic.main.fragment_create_identity_header.*

class CreateIdentityHeaderFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_create_identity_header, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        updateUI(null)
    }

    fun updateUI(name: String?, secondLine: String = "", thirdLine: String = "") {
        if (name == null || name.isEmpty()) {
            logo_layout.visibility = View.VISIBLE
            initial_layout.visibility = View.GONE
            layout_top_textlines.visibility = View.GONE
        } else {
            logo_layout.visibility = View.GONE
            layout_top_textlines.visibility = View.VISIBLE
            initial_layout.visibility = View.VISIBLE

            tv_email.setOrGone(secondLine)
            tv_phone.setOrGone(thirdLine)

            tv_name.text = name
            tv_initial.text = getInitials(name)

            tv_initial.forceLayout()
            initial_layout.forceLayout()
        }
    }

    private fun getInitials(name: String): String =
            name.split(" ".toRegex()).take(2).map {
                it.take(1).toUpperCase()
            }.joinToString("")
}
